contracts/utils/RetireToucanCarbon.sol

```diff
@@ -0,0 +1,1625 @@
+        emit PoolRouterChanged(_toucanPool, oldRouter, poolRouter[_toucanPool]);
+        return true;
+    }
+
+    /**
+        @notice Add a new carbon pool to retire with helper contract
+        @param _toucanPool Pool being added
+        @param _router UniswapV2 router to route trades through for non-pool retirements
+        @param _bondDepository Bond depository address
+        @return bool
+     */
+    function addPool(
+        address _toucanPool,
+        address _router,
+        address _bondDepository
+    ) external onlyOwner returns (bool) {
+        bool result;
+
+        require(!listContains(toucanPools, _toucanPool), "Pool already added");
+        require(_toucanPool != address(0), "Pool cannot be zero address");
+
+        toucanPools.push(_toucanPool);
+        isToucanPool[_toucanPool] = true;
+        poolRouter[_toucanPool] = _router;
+        bondDepository[_toucanPool] = _bondDepository;
+
+        emit PoolAdded(_toucanPool, _router, _bondDepository);
+        return true;
+    }
+
+    /**
+        @notice Update the bond depository for a carbon pool
+        @param _toucanPool Pool being updated
+        @param _bondDepository New depository address
+        @return bool
+     */
+    function updateBondDepository(address _toucanPool, address _bondDepository)
+        external
+        onlyOwner
+        returns (bool)
+    {
+        require(isToucanPool[_toucanPool], "Not a Toucan Carbon Pool");
+
+        address oldBond = bondDepository[_toucanPool];
+        bondDepository[_toucanPool] = _bondDepository;
+        emit PoolBondChanged(_toucanPool, oldBond, bondDepository[_toucanPool]);
+        return true;
+    }
+
+    /**
+        @notice checks array to ensure against duplicate
+        @param _list address[]
+        @param _token address
+        @return bool
+     */
+    function listContains(address[] storage _list, address _token)
+        internal
+        view
+        returns (bool)
+    {
+        for (uint256 i = 0; i < _list.length; i++) {
+            if (_list[i] == _token) {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    /**
+        @notice Call the Uniswap routers for needed amounts on token being retired.
+        @param _purchaseToken Address of token being used to purchase the pool token.
+        @param _toucanPool Address of pool token being used.
+        @param _poolAmount Amount of tokens being retired.
+        @return bool
+     */
+    function getNeededBuyAmount(
+        address _purchaseToken,
+        address _toucanPool,
+        uint256 _poolAmount
+    ) public view returns (uint256, address[] memory) {
+        address[] memory path = new address[](2);
+
+        path[0] = _purchaseToken;
+        path[1] = _toucanPool;
+
+        uint256[] memory amountIn = IUniswapV2Router02(poolRouter[_toucanPool])
+            .getAmountsIn(_poolAmount, path);
+
+        // Account for .1% default AMM slippage.
```

Comment:
Is this the fee charged by the AMM that is paid out to LPs? If so, that's .3% and already accounted for by `getAmountsIn`.